server/testserver.go

```diff
@@ -120,15 +120,32 @@ func (ts *TestServer) Start() error {
 		return util.Errorf("could not init server: %s", err)
 	}
 
-	if ts.Engine == nil {
-		ts.Engine = engine.NewInMem(proto.Attributes{}, 100<<20)
+	// Ensure we have the correct number of engines. Add in in-memory ones where
+	// needed.  There must be at least one stone (and hence engine).
+	if ts.StoresPerNode < 1 {
+		ts.StoresPerNode = 1
 	}
-	ts.Ctx.Engines = []engine.Engine{ts.Engine}
+	if ts.Engines == nil {
+		ts.Engines = make([]engine.Engine, ts.StoresPerNode)
+	}
+	ts.Ctx.Engines = make([]engine.Engine, ts.StoresPerNode)
```

Comment:
It's weird to me that we write the engines back out to the context, but assuming that's something we need here why not just set `ts.Ctx.Engines = ts.Engines` after creating the engines (or use `append(nil, ts.Engines)` to make a copy)